### DISCUSSION TITLE
Add html attributes option to grid columns

### DISCUSSION THREAD

**Discussion by tom-combet** [PrestaShop/PrestaShop#33425] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

As a developer, I want to add attributes to grid columns so that I can customize my grids independently of the data inside them.

### Describe the solution you'd like

It seems that there is no way to modify columns headers individually in a grid (see [headers_row.html.twig](https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/Table/headers_row.html.twig)). We can't override either the headers_row template (and it wouldn't be the best solution anyway imo).

Some styling attributes (like [attr](https://symfony.com/doc/current/reference/forms/types/text.html#attr) in SF Form Types) in the columns options in the grid definition could be handy, what do you think?

### Alternatives you've considered

We could make headers_row overridable.
We could also add data attributes with column name to column headers and style them wtih CSS.

These alternatives look like workarounds for me but they are fine too.

### Additional context

![image](https://user-images.githubusercontent.com/48719560/162565199-ea44979c-2a38-421c-925d-23f48ce41787.png)

Look at CMS pages grid for example. The ID column is way too large, same for position imo.
These columns will never be as large as description column, so it's useless to make them this big to me.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @tom-combet 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

PS: ping @NeOMakinG could you please check this feature :pray: 

Thank you

**Comment by NeOMakinG** [comment]
I guess most of this is related to the backend things, I couldn't do it alone!

**Comment by tom-combet** [comment]
I could do a pull request easily, but I'd like to have the opinion of the maintainers beforehand.

**Comment by kpodemski** [comment]
I like the idea. If you @tom-combet, or anybody else is motivate to work on this, feel free to do that 🎉
